### ISSUE TITLE
Building front end assets

### ISSUE THREAD

**Issue by JamesRamm** [JamesRamm/longclaw#16]
_Issue has been renamed & moved to discuss building the front end assets_

**Comment by alexfromvl** [comment]
Now admin page order/detail  has problems.

There are not enough files on the page:
1. http://..../static/longclaw.longclaworders/js/orders.bundle.js
2. http://..../static/longclaw.longclaworders/js/orders.bundle.js

**Comment by JamesRamm** [comment]
@alexfromvl You probably need to compile the JS assets. This is done automatically when running `setup.py` (e.g. when installing from github) or you can `cd` into the `longclaw/client` directory and run `npm install` followed by `npm run build`

**Comment by alexfromvl** [comment]
@JamesRamm Yes, after the manual startup files appear.

A suggestion is to make the collector start automatically when the command is "longclaw start mysite".

What do you think?

**Comment by JamesRamm** [comment]
@alexfromvl 
Building an sdist with `setup.py` will build the frontend assets so that the first distribution of longclaw submitted to pypi will have the assets already when a user `pip install`'s longclaw. You are basically coming up against this problem since longclaw has not yet been released :)
For developers of longclaw, I think it is OK to expect them to run `npm run build` since they will need to do this if they make any changes to front end components anyway.

So, long story short: When longclaw 0.1.0 is released, the assets will be included

**Comment by alexfromvl** [comment]
If I run the installation `setup.py sdist`, I get an error:

```
running sdist

> client@1.0.0 build /home/longclaw/longclaw/client
> webpack --config webpack.config.js

sh: 1: webpack: not found
```

Perhaps before starting "'npm run build", it is necessary to do the init npm "npm init"

**Comment by JamesRamm** [comment]
@alexfromvl 
you would need to do an `npm install` (in the `longclaw/client` directory) before running `npm run build`. This should install all dependencies from `package.json` for building the assets, including webpack

**Comment by JamesRamm** [comment]
@alexfromvl there is a now a command in the longclaw CLI to build the assets for you::
    ``longclaw build``
